core/commands/swarm.go

```diff
@@ -384,14 +383,13 @@ ipfs swarm connect /ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3
 			return
 		}
 
-		snet, ok := n.PeerHost.Network().(*swarm.Network)
+		// FIXME(steb): Nasty
+		swrm, ok := n.PeerHost.Network().(*swarm.Swarm)
```

Comment:
ah, because we need to expose the backoff logic... It probably makes sense to just embed that information in the context.

Reply:
```go
ctx := context.WithValue(ctx, "swarmClearBackoff", true)
n.Peerhost.Connect(ctx, ...)
```

Reply:
though, downside there is that it becomes much harder to notice when it stops working.

Reply:
For now, I'm planning on leaving this as it is. This is more for my future self.